### DISCUSSION TITLE
Website renovation

### DISCUSSION THREAD

**Discussion by cooldude6000** [mesa/mesa#2063] [Ideas]
I want to update the https://mesa.readthedocs.io/en/latest/ by adding MANY new features and making it attractive and if permitted, I would more than love to rebuild an entire website for MESA along with docs and animated code templates for new users to try/test out.
Let me know the opinions of the mentors.

**Comment by rht** [comment]
Improving the docs sounds good to me. Do you mind to elaborate a bit further? Though, regarding with the animated code templates, they can't be copy-pasted, and new users might have a typo problem.

**Reply by cooldude6000** [reply]
Hey,
Regarding the animated code snippets: In the model that I am proposing, the animated code CAN be copy-pasted and the users will not have any problem rather than appreciating the websites' UI.
I made an example code snippet for you to look at : https://codepen.io/Gowreesh-Simhadri/pen/dyLyOVv

Website Enhancements:

User Interface and Functionality:
•Implement user login and logout features for authorized administrators to manage the website, specifically for posting blog content.
•Ensure responsive design across mobile, tablet, and laptop devices for seamless user experience on any platform.
•Consider incorporating modern design elements, using neon colors cautiously and strategically, to create a more visually engaging website.
•Integrate a statistical dashboard displaying user engagement metrics and positive feedback to showcase the website's impact.

Documentation Improvement:
•Develop a comprehensive "How to Get Started" guide to aid new contributors and users.
•Enhance the documentation with all necessary information, ensuring clarity and ease of access.
•Integrate a dedicated "Documentation" tab within the website for easy access to all relevant information.

Accessibility Features:
•Implement multi-language support for the website, including French, Spanish, German, and other relevant languages to expand its reach and inclusivity.
•Integrate search functionality within both the website and documentation for efficient information retrieval.
•Explore the feasibility of incorporating animated code snippets within blog posts for interactive learning and engagement.


I also have quite a few other ideas to further enhance the website.

**Reply by EwoutH** [reply]
Thanks for your proposal!
- I don't see the value in the website. If we every want to do blogs I imagine that will just happen though git and Readthedocs or maybe Github pages. I don't want an login system to maintain.
- A serious improvement in the tutorial would be welcome. But that does require a lot of ABM specific knowledge.
- Please be more specific than "Enhance the documentation with all necessary information"
- Multi-language would be useful
- Search would be useful (but we have [that](https://mesa.readthedocs.io/en/stable/))
- Easier animations would be useful.

In general, good ideas, but I see a lot of solutions without corresponding problems. Also please consider it's effect on maintenance.

Currently I don't see a lot of value in a separate website aside from our ReadtheDocs. I would support efforts to improve the ReadtheDocs, but be very specific in what problems you want to solve.

**Reply by rht** [reply]
> Regarding the animated code snippets: In the model that I am proposing, the animated code CAN be copy-pasted and the users will not have any problem rather than appreciating the websites' UI.
> I made an example code snippet for you to look at : https://codepen.io/Gowreesh-Simhadri/pen/dyLyOVv

I saw it. Interactivity seems to make sense if the code to be typed is a CLI that is meant to be brief and step-by-step, e.g. how to set up GPU on non-NVIDIA devices for llama.cpp.

Before we translate to various languages, I think the English documentation is in need for a revamp first.
> •Develop a comprehensive "How to Get Started" guide to aid new contributors and users.
> •Enhance the documentation with all necessary information, ensuring clarity and ease of access.

Regarding with the structure of the documentation, I'm thinking of basing on https://diataxis.fr/ (@jackiekazil found out about this during PyCon last year). Several projects' structure are based on Diataxis, including Solara.

> •Integrate search functionality within both the website and documentation for efficient information retrieval.

A fancy feature that I have in mind is to have an LLM chat interface to the docs, e.g. https://chat.langchain.com/ for LangChain. So that users won't have to go to ChatGPT for conversational questions.

**Reply by cooldude6000** [reply]
I'm sorry you felt it that way. I thought the looks of website REALLY do matter, so I bought this idea of renovating/modernising the website from scratch and adding the features to the UI to make it attractive and interactive to the users.

Thanks for your time and reply.

**Reply by jackiekazil** [reply]
Agree with what has been said. Thank you for bringing this up @gowreesh36, I think it has given us some things to prioritize and consider.
